agent/stats/utils_windows.go

```diff
@@ -31,10 +31,14 @@ func dockerStatsToContainerStats(dockerStats *types.StatsJSON) (*ContainerStats,
 	cpuUsage := (dockerStats.CPUStats.CPUUsage.TotalUsage * 100) / numCores
 	memoryUsage := dockerStats.MemoryStats.PrivateWorkingSet
 	networkStats := getNetworkStats(dockerStats)
+	storageReadBytes := dockerStats.StorageStats.ReadSizeBytes
```

Comment:
It would be better to check `if dockerStats.StorageStats != nil` , else this could break. 
same for linux (just as a safeguard - sorry missed this earlier)

Reply:
We will need to add this all over the place.  Just in this file we also have `dockerStats.CPUStats.CPUUsage.TotalUsage`, `dockerStats.MemoryStats.PrivateWorkingSet`.  These have been in place for a while now with no checks.  Should we expect that the StorageStats object potentially will not exist in the docker stats?

Reply:
I'm not sure. I have seen networkStats to be nil in some cases. If you're sure storage stats will never be empty we can leave this as is.

Reply:
Will need to research which stats fields can be nil (which aren't required).

Reply:
we can revisit it outside the PR